usaspending_api/common/query_with_filters.py

```diff
@@ -0,0 +1,426 @@
+import logging
+
+from django.conf import settings
+
+from abc import abstractmethod, ABCMeta
+from typing import Union, List
+from elasticsearch_dsl import Q as ES_Q
+
+from usaspending_api.common.exceptions import InvalidParameterException
+
+
+logger = logging.getLogger(__name__)
+
+
+class _Filter(metaclass=ABCMeta):
+    """
+    Represents a filter object used to currently query only Elasticsearch.
+    """
+
+    underscore_name = None
+
+    @classmethod
+    def generate_query(cls, filter_values: Union[str, list]) -> dict:
+
+        if filter_values is None:
+            raise InvalidParameterException(f"Invalid filter: {cls.underscore_name} has null as its value.")
+
+        return cls._generate_elasticsearch_query(filter_values)
+
+    @classmethod
+    @abstractmethod
+    def _generate_elasticsearch_query(cls, filter_values: Union[str, list]) -> Union[ES_Q, List[ES_Q]]:
+        """ Returns a Q object used to query Elasticsearch. """
+        raise NotImplementedError("Must be implemented in subclasses of _Filter.")
```

Comment:
Is this exception redundant when using `@abstractmethod`?

Reply:
Replaced with `pass` because it is redundant.